addons/account_edi_extended/models/account_edi_document.py

```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo import models, fields, _
+import logging
+
+_logger = logging.getLogger(__name__)
+DEFAULT_BLOCKING_LEVEL = 'warning'  # Keep previous behavior. TODO : when account_edi_extended is merged with account_edi, should be 'error' (document will not be processed again until forced retry or reset to draft)
+
+
+class AccountEdiDocument(models.Model):
+    _inherit = 'account.edi.document'
+
+    blocked_level = fields.Selection(selection=[('info', 'Info'), ('warning', 'Warning'), ('error', 'Error')],
+                                     help="Blocks the document current operation depending on the error severity :\n"
+                                          "  * Info: the document is not blocked and everything is working as it should.\n"
+                                          "  * Warning : there is an error that doesn't prevent the current Electronic Invoicing operation to succeed.\n"
+                                          "  * Error : there is an error that blocks the current Electronic Invoicing operation.")
```

Comment:
Ok to change the name.

Yes, it can be None if the invoice is not blocked ('error' is only the text of the error but the real business field is this one)

Reply:
But `info` and `warning` are non-blocking, so why put `False`?

Reply:
In the views, we display different things/things differently based on that field (or its aggregated version for the invoice 'edi_error_level'). I guess we could find a way to make the field required, but IMO it would make it more confusing and possibly result in bad coding in the future when we add new features (I prefer that if there are no error nor warning nor info, then the field is False).